app/lib/backend_api_logic/routing_policy.rb

```diff
@@ -18,7 +18,7 @@ def policy_chain
     end
 
     def with_subpaths?
-      backend_api_configs.with_subpath.any?
+      backend_api_configs.any?(&:with_subpath?)
```

Comment:
This was already merged with #3845 

If you look from the perspective of taking a service and calling this method on it, what you say makes sense. I fixed a number of N+1 though with the aforementioned PR and I don't remember which test it was related to. Also I looked at a couple of requests and optimized them to a maximum least number of queries.

I think in this case, this change is based on the premises that we have the backend_api_configs already preloaded for the service(s) we deal with. So calling `with_subpath?` on each is more effective than performing a new database query. Applying a scope results in a new query.

In the other PR I have added `active_record_query_trace` gem and I assume it showed to me that a query came from this line where I didn't expect a query at this point.

I think even if backend_api_configs is not preloaded, it wouldn't be a huge deal because I don't expect too many backends in services. And it will still be one query, although with more data returned than the original code. It will also load the backend_api_configs into the respective service in case they are further needed.

If you have spotted a particular call that is less efficient this way, we may think about it. But I think it might likely be an edge case when a single service is involved. But with original code I don't see how we can avoid N+1 when many services are loaded at once and we want to preload everything needed for presentation. As the original code will still try to perform a new query for each service.

Hope explanation makes sense. But better comment further on the original (merged) PR because I will be rebasing this one to avoid all the extra commits already in master.

Reply:
Nice, so data is preloaded and you save a query. Fine.